[MRG] Changed the number of features from 10 to variable

map222 opened the pull request:
I ported this code into my own to try to understand feature importance. When I tried plotting my features, I found the number of features was hard coded to 10. Here I make n_features a variable. By switching to a set variable, the number of features is no longer a magic number, and the code is more readable.

map222 committed:
Changed the number of features from 10 to variable

I ported this code into my own, and when I tried plotting, I found the number of features was hard coded to 10. By switching to a set variable, the number of features is no longer a magic number, and the code is more readable.

jnothman commented:
thanks, lgtm!

jnothman commented:
Actually, I've changed my mind. Can you use `X.shape[1]` in all the places following X's definition? Then the user can substitute their own dataset. You don't need a separate n_features variable then.

map222 commented:
OK, I can make that change later tonight.
On Oct 11, 2015 17:32, "Joel Nothman" <notifications@github.com> wrote:

> Actually, I've changed my mind. Can you use X.shape[1] in all the places
> following X's definition? Then the user can substitute their own dataset.
> You don't need a separate n_features variable then.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/scikit-learn/scikit-learn/pull/5384#issuecomment-147266277>
> .
>

map222 committed:
Now plots based on the number of features in X

In previous commit,  I used n_features to set the number of features, and plotted based on that variable. Based on feedback, I removed n_features, and now plotting is based on X.shape[1]. This makes the code general and easy to port. I copied this into an iPython notebook to verify it still outputs the correct graph.

ogrisel commented:
LGTM as well. Merged.

ogrisel commented:
Thanks @map222!
